frontend/context/StakingContractDetailsProvider.tsx

```diff
@@ -0,0 +1,181 @@
+import { useQueries, useQuery } from '@tanstack/react-query';
+import {
+  createContext,
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useCallback,
+  useContext,
+  useState,
+} from 'react';
+
+import { AGENT_CONFIG } from '@/config/agents';
+import { GNOSIS_CHAIN_CONFIG } from '@/config/chains';
+import { FIVE_SECONDS_INTERVAL } from '@/constants/intervals';
+import { REACT_QUERY_KEYS } from '@/constants/react-query-keys';
+import { StakingProgramId } from '@/enums/StakingProgram';
+import { useService } from '@/hooks/useService';
+import { useServices } from '@/hooks/useServices';
+import { StakingContractDetails } from '@/types/Autonolas';
+
+import {
+  INITIAL_DEFAULT_STAKING_PROGRAM_ID,
+  StakingProgramContext,
+} from './StakingProgramProvider';
+
+const currentAgent = AGENT_CONFIG.trader; // TODO: replace with dynamic agent selection
+const currentChainId = GNOSIS_CHAIN_CONFIG.chainId; // TODO: replace with dynamic chain selection
+
+// TODO: create a separate hook to get service id
+const useServiceId = () => {
+  const { selectedService, isFetched: isLoaded } = useServices();
+  const serviceConfigId =
+    isLoaded && selectedService ? selectedService?.service_config_id : '';
+  const { service } = useService({ serviceConfigId });
+  const serviceId = service?.chain_configs[currentChainId].chain_data?.token;
+
+  return serviceId;
+};
```

Comment:
this doesn't need a hook imo :/
we can get `service` from `useService` and access the id from there